src/plugins/utils/performance-utils.ts

```diff
@@ -1,17 +1,30 @@
-export const defaultPerformanceIgnore = (entry: PerformanceEntry) =>
+import { ResourceType } from '../../utils/common-utils';
+
+export const defaultIgnore = (entry: PerformanceEntry) =>
     entry.entryType === 'resource' && !/^https?:/.test(entry.name);
 
 export type PartialPerformancePluginConfig = {
-    ignore?: (event: PerformanceEntry) => any;
     eventLimit?: number;
+    ignore?: (event: PerformanceEntry) => any;
+    recordAllTypes?: ResourceType[];
+    sampleTypes?: ResourceType[];
 };
 
 export type PerformancePluginConfig = {
-    ignore: (event: PerformanceEntry) => any;
     eventLimit: number;
+    ignore: (event: PerformanceEntry) => any;
+    recordAllTypes: ResourceType[];
```

Comment:
issue: we need to update the performance config documentation

Reply:
Documentation for `eventLimit`, `recordAllTypes` and `sampleTypes` has never existed, so there's no regression or omission with respect to this PR.

Let's close out this PR and track that separately.